express-notes-api/server.js

```diff
@@ -0,0 +1,145 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+const app = express();
+
+app.use(express.json());
+
+async function readNotes() {
+  try {
+    const data = await readFile('data.json', { encoding: 'utf8' });
+    const jsonData = JSON.parse(data);
+    return jsonData;
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+async function writeNotes(notesData) {
+  try {
+    const jsonData = await readFile('data.json', { encoding: 'utf8' });
+    const data = JSON.parse(jsonData);
+
+    data.notes = notesData.notes;
+    data.nextId = notesData.nextId;
+
+    await writeFile('data.json', JSON.stringify(data, null, 2), { encoding: 'utf8' });
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+app.get('/api/notes', async (req, res) => {
+  try {
+    const jsonData = await readNotes();
+    const notesData = jsonData.notes;
+    const notesArray = [];
+    for (const id in notesData) {
+      notesArray.push(notesData[id]);
+    }
+    res.status(200).send(notesArray);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const id = req.params.id;
+  if (isNaN(id) || id <= 0) {
+    res.status(400).json({ error: 'Id must be a positive integer' });
+  }
```

Comment:
Please improve your id validation to send `400 Bad Request` responses on positive non-integer numbers.